src/featurelistmodel.cpp

```diff
@@ -156,18 +167,67 @@ void FeatureListModel::reloadLayer()
 
   QgsFeature feature;
 
-  QStringList displayStrings;
-  QVariantList keys;
+  QList<Entry> entries;
+
+  if ( mAddNull )
+    entries.append( Entry( QStringLiteral( "<i>NULL</i>" ), QVariant( QVariant::Int ) ) );
 
   while ( iterator.nextFeature( feature ) )
   {
     context.setFeature( feature );
-    displayStrings.append( expression.evaluate( &context ).toString() );
-    keys.append( feature.attribute( keyIndex ) );
+    entries.append( Entry( expression.evaluate( &context ).toString(), feature.attribute( keyIndex ) ) );
+  }
+
+  if ( mOrderByValue )
+  {
+    qSort( entries.begin(), entries.end(), [] ( const Entry &entry1, const Entry &entry2 )
+    {
+      if ( entry1.key.isNull() )
```

Comment:
Confused about that. Shouldn't it check if `displayString` is null?

Reply:
It's to have the `NULL` entry on top, where the key is a Null QVariant and the text is `<i>NULL</i>`

Reply:
Ah, that makes sense :-)